figures/scripts/figS6_UMAPs_seq-center.R

```diff
@@ -0,0 +1,119 @@
+# S. Spielman 2022
+# Script to make UMAPs exploring potential sequencing center batch effects
+
+# Libraries --------
+library(magrittr)
+
+# Define paths and files ---------------------
+dim_red_dir <- file.path(
+  "analyses",
+  "transcriptomic-dimension-reduction"
+)
+
+metadata_file <- file.path("data", "pbta-histologies.tsv")
+pal_file <- file.path("figures", "palettes", "broad_histology_cancer_group_palette.tsv")
+rsem_umap_file <- file.path(
+  dim_red_dir,
+  "results",
+  "rsem_stranded_log_umap_scores_aligned.tsv"
+)
+
+
+# Output:
+output_dir <- file.path("figures", "pdfs", "supp", "figS5", "panels")
+umap_all_bh_pdf <- file.path(output_dir, "umap_sequencing_all_histologies.pdf")
+umap_shared_bh_pdf <- file.path(output_dir, "umap_sequencing_shared_histologies.pdf")
+umap_centers_only_pdf <- file.path(output_dir, "umap_sequencing_centers.pdf")
+
+
+# Source the UMAP plotting function -------------
+source(file.path(dim_red_dir, "util", "dimension-reduction-functions.R"))
+
+
+# Read in data files ----------------
+histologies_df <- readr::read_tsv(metadata_file, guess_max = 100000)
+
+rsem_umaps <- readr::read_tsv(rsem_umap_file)
+
+palette_mapping_df <- readr::read_tsv(pal_file) %>%
+  dplyr::select(broad_histology, broad_histology_display, broad_histology_hex) 
+
+
+# Define the color key -----------------
+broad_histology_color_df <- palette_mapping_df %>%
+  dplyr::select(broad_histology_display, broad_histology_hex) %>%
+  dplyr::distinct() 
+annotation_colors_bh<- broad_histology_color_df$broad_histology_hex
+names(annotation_colors_bh) <- broad_histology_color_df$broad_histology_display
+
+
+# Render UMAPs ----------------------
+
+# First, link IDs to histology display
+palette_mapping_df <- palette_mapping_df %>%
+  dplyr::inner_join(histologies_df, by="broad_histology") %>%
+  dplyr::select(Kids_First_Biospecimen_ID, broad_histology_display)
+
+# Prepare data for UMAP plot
+umap_df <- rsem_umaps %>%
+  dplyr::select(Kids_First_Biospecimen_ID, seq_center, X1, X2) %>%
+  # Join on color palette info
+  dplyr::left_join(palette_mapping_df,
+                    by = "Kids_First_Biospecimen_ID") %>%
+  dplyr::select(X1, X2, seq_center, broad_histology_display) 
+
+# Shuffle the rows going in for alpha purposes, and
+#  set order for `seq_center` as group size
+set.seed(2022)
+umap_df <- umap_df %>%
+  dplyr::sample_n(nrow(umap_df)) %>%
+  dplyr::mutate(seq_center = forcats::fct_infreq(seq_center))
+
+# UMAP of all histologies
+umap_all_bh <- plot_dimension_reduction(umap_df,
+                                        point_color = "broad_histology_display",
+                                        point_shape = "seq_center",
```

Comment:
For this plot, I think it might be more clear to have the color by sequencing center and the broad histology shapes. Or you could just incorporate the shapes into the later plot where you color by sequencing center? I just found it hard to pick out the sequencing centers on this plot.

Reply:
I agree that it's hard to pick out the centers. Something else I think will help is using outlined points w/ color and fill for low N sequencing centers, which means not using the plot wrapper (which is very fine).